js/okex.js

```diff
@@ -790,69 +790,70 @@ module.exports = class okex extends Exchange {
-                const optionType = this.safeString (market, 'optType');
+                strikePrice = this.safeString (market, 'stk');
+                optionType = this.safeString (market, 'optType');
                 symbol = symbol + '-' + strikePrice + '-' + optionType;
+                optionType = (optionType === 'P') ? 'put' : 'call';
             }
         }
         const tickSize = this.safeString (market, 'tickSz');
-        const precision = {
-            'amount': this.safeNumber (market, 'lotSz'),
-            'price': this.parseNumber (tickSize),
-        };
         const minAmountString = this.safeString (market, 'minSz');
         const minAmount = this.parseNumber (minAmountString);
         let minCost = undefined;
         if ((minAmount !== undefined) && (tickSize !== undefined)) {
             minCost = this.parseNumber (Precise.stringMul (tickSize, minAmountString));
         }
-        const active = true;
         const fees = this.safeValue2 (this.fees, type, 'trading', {});
-        let contractSize = undefined;
-        if (contract) {
-            contractSize = this.safeNumber (market, 'ctVal');
-        }
-        const leverage = this.safeNumber (market, 'lever', 1);
+        const precisionPrice = this.parseNumber (tickSize);
+        let maxLeverage = this.safeString (market, 'lever', '1');
+        maxLeverage = Precise.stringMax (maxLeverage, '1');
         return this.extend (fees, {
             'id': id,
             'symbol': symbol,
             'base': base,
             'quote': quote,
+            'settle': settle,
             'baseId': baseId,
             'quoteId': quoteId,
-            'settleId': settleCurrency,
-            'settle': settle,
-            'info': market,
+            'settleId': settleId,
             'type': type,
             'spot': spot,
-            'futures': futures,
+            'margin': spot && (Precise.stringGt (maxLeverage, 1)),
             'swap': swap,
-            'contract': contract,
+            'futures': futures,
             'option': option,
-            'linear': linear,
-            'inverse': inverse,
-            'active': active,
-            'contractSize': contractSize,
-            'precision': precision,
+            'active': true,
+            'contract': contract,
+            'linear': contract ? (quoteId === settleId) : undefined,
+            'inverse': contract ? (baseId === settleId) : undefined,
+            'contractSize': contract ? this.safeNumber (market, 'ctVal') : undefined,
             'expiry': expiry,
             'expiryDatetime': this.iso8601 (expiry),
+            'strike': strikePrice,
+            'optionType': optionType,
+            'precision': {
+                'price': precisionPrice,
+                'amount': this.safeNumber (market, 'lotSz'),
+            },
             'limits': {
+                'leverage': {
+                    'min': 1,
```

Comment:
We need `this.parseNumber ('1')` in all such cases for `min`, cause we're using `parseNumber` for `max` as well.

Reply:
So what are the rules for using `parseNumber` because I saw that you removed it here https://github.com/ccxt/ccxt/pull/11586/commits/a51af12e689b4dc436792f483f3891c28ef47416#r793046692

Reply:
@samgermain i removed it from `precision` with default `precisionMode = DECIMAL_DIGITS`, that's ok, but we should still use parseNumber for `limits`.

Reply:
I don't really understand what you mean by "default `precisionMode = DECIMAL_DIGITS`"

This is the only place I can find `parseNumber` in the contributing guidelines ![](https://i.imgur.com/fnBGbo2.png)
I can change it here, but I don't really understand when to use `parseNumber` and when not to use it

Reply:
> I don't really understand what you mean by "default `precisionMode = DECIMAL_DIGITS`"

Each exchange has this property called `exchange.precisionMode` that drives the precision for markets and currencies. Most of the exchanges have `exchange.precisionMode = DECIMAL_DIGITS`, which is why `DECIMAL_DIGITS` is the default value of `exchange.precisionMode`. The exchanges that have a different `precisionMode` – override it in their `describe()` implementation. This is covered here:

- https://github.com/ccxt/ccxt/wiki/Manual#precision-mode

When `precisionMode = DECIMAL_DIGITS`, then the values inside `precision` are expected to be **integer numbers** of decimal digits after the dot for the corresponding amounts and prices. For example, you cannot have 4.5 decimal digits after the dot, it's either 4 digits or 5 digits. Hence, the default `precisionMode = DECIMAL_DIGITS` effectively means _precision as integers_.

> I can change it here, but I don't really understand when to use `parseNumber` and when not to use it.

When the value is an integer – (a timestamp or a number of countable things, like a number of decimal digits in `precision`) – don't use parseNumber.

When the value being parsed is a fraction of a currency – then use parseNumber.

Reply:
On a sidenote, this specific exchange, okex v5, uses precisionMode = TICK_SIZE, so we should use parseNumber in `precision`.

Reply:
Does anything need to be changed or is the current implementation good enough

Reply:
I think it's good now, thx.